common/v2/features/Dashboard/components/WalletBreakdown/WalletBreakdownView.tsx

```diff
@@ -229,15 +228,22 @@ export default function WalletBreakdownView({
         $total: `${accounts.length}`
       });
 
-  const shownSelectedIndex =
-    chartBalances.length > selectedAssetIndex && selectedAssetIndex !== -1 ? selectedAssetIndex : 0;
+  const shownSelectedIndex = !(
+    chartBalances.length > selectedAssetIndex && selectedAssetIndex !== -1
+  )
+    ? chartBalances.length <= selectedAssetIndex
+      ? chartBalances.length - 1
+      : 0
+    : selectedAssetIndex;
+
```

Comment:
Nested ternary are hard to read and harder to maintain.
Please extract into a function and include a unit test to make sure we preserve the desired behaviour.